pkg/providers/vsphere/vmprovider_resourcepolicy.go

```diff
@@ -58,7 +59,7 @@ func (vs *vSphereVMProvider) CreateOrUpdateVirtualMachineSetResourcePolicy(
 			clusterRef, err := vcenter.GetResourcePoolOwnerMoRef(ctx, vimClient, rpMoID)
 			if err == nil {
 				clusterModuleProvider := clustermodules.NewProvider(client.RestClient())
-				err = vs.createClusterModules(ctx, clusterModuleProvider, clusterRef.Reference(), resourcePolicy)
+				err = vs.reconcileClusterModules(ctx, clusterModuleProvider, clusterRef.Reference(), resourcePolicy)
```

Comment:
This will miss the case when all the cluster modules are removed, no?

Reply:
Refactored, removed the if and also moved it out of this loop (see below)